Issue-35 branch, squashed, rebased and slightly cleaned

frankvanbever opened the pull request:
There were a lot of work-in-progress artifacts on the Issue-35 branch so I decided to squash all commits. I did a first pass through the code and cleaned up some things. The code for the alarms is very dense. I think it would benefit from a short introduction in the docstring at the beginning of every function instead of comments for every line.

Can someone check if alarm generation is still working before merging this?

I'm also regularly getting the following in the output:

```
[WARNING] Exception occurred:  'Cursor' object has no attribute 'find'
```

erademakers committed:
Add Processing thread to determine alarms

- added dummy print from file function + dataset
- add exception in case the data doesn't contain peaks in the pressure
- add exception in case the data doesn't contain peaks in the pressure
- each process must have its own instance of the mongodb, so added also a
  client instance in the database processing and moved the database read
  functions to the database processing class, where they are used
- added the overshoot peak pressure
- fix indexing with numpy arrays
- added function to detect when the pressure peak is outside the allowed range
- added another MonitorVolume class + volume peak not in allowed range function
- this class will contain the processing functions required to analyze the
  volume data and trigger the alarms
- function to detect when volume not near zero at ebc
- function to check volume near 0 ==>> at the end of every cycle
  return the following
  * nbr of time volume was not near zero
  * threshold set to 5 as default value

Note: the volume  monitor class function are not tested yet on any data

Squashed and cleaned up from the original branch

Signed-off-by: Frank Vanbever <frank.vanbever@essensium.com>

Merwanski commented:
Hey @frankvanbever 
Just rechecked the code there was an error at line 300 when the volume data are requested
```
elif type_data == 'VOL':
            return self.db.volume_values.find().find().sort("loggedAt", -1).limit(N)
```
the function ".find()" is called two times 
You can try it now again !! 
PS: if you are using dummyPrint_from_file script with the volume part you will still get a WARNING and an Alarm because the JSON file has no volume data In it 

Should I fix this in the issue-35 branch or you do it in the master branch?

frankvanbever committed:
database_processing: Remove superfluous find()

Signed-off-by: Frank Vanbever <frank.vanbever@essensium.com>

frankvanbever committed:
database_processing: Use proto.alarm to signal

Also cleaned up the imports in database_processing

Signed-off-by: Frank Vanbever <frank.vanbever@essensium.com>

frankvanbever committed:
ventilator_alarm: don't put values on alarm queue

Given that analysis is done on values retrieved from the data base we no longer
need to have these sent to the alarm queue.

Signed-off-by: Frank Vanbever <frank.vanbever@gmail.com>

frankvanbever committed:
ventilator_alarm: remove hardcoded "ALARM" comparison

Using the definition in proto allows us to easily change it should it become
necessary.

Signed-off-by: Frank Vanbever <frank.vanbever@gmail.com>

frankvanbever commented:
@Merwanski I put the fix on here

frankvanbever committed:
ventilator_alarm: retain old alarm and propagate

We update the Controller about the alarm state every second, but for good
measure we put an additional alarm on the queue as soon as we get it. To not
load the UI only send an alarm update if the alarm state changes.

Signed-off-by: Frank Vanbever <frank.vanbever@gmail.com>
